lib/validator.js

```diff
@@ -89,32 +89,31 @@ function* checkInterfaceMemberDuplication(defs) {
   }
 
   function* forEachInterface(i) {
-    const opNames = getOperationNames(i);
+    const opNames = new Set(getOperations(i).map(op => op.name));
     const partials = defs.partials.get(i.name) || [];
     const mixins = includesMap.get(i.name) || [];
-    for (const partial of partials) {
-      yield* forEachExtension(partial, opNames, i);
-    }
-    for (const mixin of mixins) {
-      yield* forEachExtension(mixin, opNames, i);
+    for (const ext of [...partials, ...mixins]) {
+      const additions = getOperations(ext);
+      yield* forEachExtension(additions, opNames, ext, i);
+      for (const addition of additions) {
+        opNames.add(addition.name);
+      }
     }
   }
 
-  function* forEachExtension(ext, names, base) {
-    for (const op of ext.members.filter(mem => mem.type === "operation")) {
-      const name = (op.body && op.body.name) ? op.body.name.value : "";
-      if (name && names.has(name)) {
-        const message = `The operation "${name}" has already been defined in the base interface "${base.name}"`;
-        yield error(ext.source, op.body.tokens.name, ext, message);
+  function* forEachExtension(additions, existings, ext, base) {
+    for (const addition of additions) {
+      const { name } = addition;
+      if (name && existings.has(name)) {
+        const message = `The operation "${name}" has already been defined for the base interface "${base.name}"`;
```

Comment:
Don't know if it helps or too long... feel free to reject: 

```suggestion
        const message = `The operation "${name}" has already been defined for the base interface "${base.name} either in the base class or in a mixin"`;
```

Reply:
Using the word `class` doesn't look good... Hmm.